### PR TITLE
Changed the behavior of new / save buttons (#86)

### PR DESCRIPTION
**Pull request by reisfmb**

### CODE REVIEW

File: src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html [hunk 1]
```diff
@@ -31,6 +31,62 @@
         };
       },
       methods: {
+        newCty(){
+          if(this.checkBrowserCompatibility()){
```
Comment (poi33) [inline src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html:35]: What if this returns false?

File: src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html [hunk 2]
```diff
@@ -31,6 +31,62 @@
...
+            this.fileHandle = null;
+            this.$store.commit("resetJson");
+            this.$store.commit("setFileName", { data: ' ' });
+          }
+        },
+        async loadCty() {
```
Comment (poi33) [inline src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html:41]: My suggestion to improve the loadCty. (Feel free to ignore) Its pseudo code so you think to fill in the gaps and create the function that don't exist.
```
async loadCty function () {
    const file = await selectFile();
    await showLoadingIcon()
    await setJson(file.json())
    await removeLoadingIcon()
}
```
Don't know if its possible to create this behavior.

File: src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html [hunk 3]
```diff
@@ -217,8 +232,9 @@
             });
           }
         },
-
+        
         ///
```
Comment (poi33) [inline src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html:236]: /// ?

File: src/main/resources/admin/tools/content-type-manager/modules/module-example.html [hunk 4]
```diff
@@ -29,34 +29,32 @@
...
-        attributes: {
-          version: "1.0",
-          encoding: "UTF-8",
-        },
-      },
-      elements: [
-        {
-          type: "element",
-          name: "content-type",
+    ///
```
Comment (poi33) [inline src/main/resources/admin/tools/content-type-manager/modules/module-example.html:32]: /// ?

File: src/main/resources/admin/tools/content-type-manager/modules/module-example.html [hunk 5]
```diff
@@ -69,7 +67,23 @@
...
+                  name: "form", 
+                  elements: [] 
+                }
+              ] 
+            } 
+          ];
+          // Used empty array + setTimeout as a quick way to 
+          // remount components and their inner empty states.
+          state.json.elements = [];
+          setTimeout(() => { state.json.elements = initialJsonElements }, 100);
```
Comment (poi33) [inline src/main/resources/admin/tools/content-type-manager/modules/module-example.html:86]: This seems hacky? Is this the best way to do it? I don't know how mouting works in vue so feel free to ignore

**Review by poi33 — CHANGES_REQUESTED** [review]

File: src/main/resources/admin/tools/content-type-manager/modules/module-example.html (hunk 4 above)
Comment (reisfmb) [reply src/main/resources/admin/tools/content-type-manager/modules/module-example.html:32]: Removed.

File: src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html (hunk 3 above)
Comment (reisfmb) [reply src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html:236]: Removed.

File: src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html (hunk 2 above)
Comment (reisfmb) [reply src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html:41]: I tried to improve the _loadCty_ function following some of your suggestions. Please take a look at the last commit :smile:

File: src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html [hunk 6]
```diff
@@ -31,6 +31,62 @@
...
+                  text: this.warnings['file-not-xml'],
+                  icon: "error",
+                  showConfirmButton: false,
+                  timer: 1500,
+                });
+                this.fileHandle = null;
+                return false;
+              }
+
+              this.setJson(await file.text());
```
[the remark this answers is not in the document]
Comment (reisfmb) [reply src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html]: Please check the updates on the _loadCty_ function.

File: src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html [hunk 7]
```diff
@@ -39,44 +39,60 @@
...
+              showConfirmButton: false,
+              timer: 1500,
+            }).then(() => {
+              this.$store.commit("setFileName", {
+                data: file.name.replace(".xml", ""),
               });
             });
           }
+
+          showSelectXmlDialog()
```
Comment (poi33) [inline src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html:81]: Nice, does it work and the loading closes when its done with converting the xml to json and generating the dom?

**Review by poi33 — APPROVED** [review]

**Review by poi33 — CHANGES_REQUESTED** [review]
Could you change the checkBroswerCompatibility? 
I don't think its a good way to check and show dom messages from inside a if test

File: src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (reisfmb) [reply src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html:35]: @poi33

"The current way that it is used I would expect it to be outside an if statment."

Why would you expect that? the _checkBrowserCompatibility_ function returns _true / false_ and also shows a dialog in the case of the false return, therefore I need to use the _true / false_ in order to proceed with the code execution.

If we use it outside of an if statement, we'll be ignoring it's result. 

For example:
```
this.checkBrowserCompatibility()
{...}
```

In this case lines 2, 3, and so on we'll be executed regardless of the compatibility of the browser, and that's not the desired behavior for the functions: newCty, loadCty and saveCty.

Makes sense? Am I missing something?

[the remark this answers is not in the document]
Comment (poi33) [reply src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html]: Here is how i would expect the checkBrowserCompatibility() to work.
```
if (checkBrowserCompatibility()) {
    // Execute all 
} else {
    showDialogMessage("You browser is not supported")
}
```

I don't like that a method i expect to only be false or true also executes code that shows a dialog

[the remark this answers is not in the document]
Comment (reisfmb) [reply src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html]: Yes, makes sense... Take a look at this [commit](https://github.com/enonic/app-cty-manager/pull/102/commits/e9b5ba91e54a97519a31e998c27d171db1c36e42) related to this PR.

[the remark this answers is not in the document]
Comment (poi33) [reply src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html]: Just for the record. If you think this is fine, and think its better the way it is. 
I don't mind closing this and we do it like this.

**Review by poi33 — APPROVED** [review]

File: src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (reisfmb) [reply src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html]: Yes, but I agree with what you've suggested!

**Pull request merged**